test/hotspot/jtreg/compiler/ccp/TestShiftConvertAndNotification.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8313672
+ * @summary Test CCP notification for value update of AndL through LShiftI and ConvI2L.
+ * @run main/othervm -XX:RepeatCompilation=1000 -XX:-TieredCompilation
+ *                   -XX:+StressIGVN -XX:-CICompileOSR -Xcomp
+ *                   -XX:CompileCommand=compileonly,TestShiftConvertAndNotification::test
+ *                   compiler.ccp.TestShiftConvertAndNotification
```

Comment:
How long does this test take to execute? `-XX:RepeatCompilation=1000` could make this test slow.

Do you need `-Xcomp`, or is `-Xbatch` sufficient? Or even without?

Also: I would add another `@run` description which runs the test with no flags, or at most `-Xcomp`. It has happened often that this triggers new bugs, maybe in combination with other flags set from the outside. And for that it is good to have a run with very few or better no flags/args at all. And the testing people like it if you don't just have multiple `@run` statements, so just duplicate the whole:

```
/*
 * @test
 * ..
 */
```

Reply:
Yes, I need to tweak this. The failure is, for some reason, not appearing when run through jtreg. I'll get back to you (also with answers to your other questions).

Reply:
Fixed now. `-XX:RepeatCompilation=20` is fast and seems enough to practically always trigger the failure (although it is non-deterministic due to `-XX:+StressIGVN`).

I have not managed to reproduce the bug without `-Xcomp`.

Running the test with only `-Xcomp` is quite slow (~1 minute on my system). I added the below instead (the test itself completes in <1 second).
```
/*
 * @test
 * @bug 8313672
 * @summary Test CCP notification for value update of AndL through LShiftI and
 *          ConvI2L (reduced set of flags).
 * @run main/othervm -XX:CompileCommand=compileonly,compiler.ccp.TestShiftConvertAndNotification::test
 *                   -Xcomp
 *                   compiler.ccp.TestShiftConvertAndNotification
 *
 */
```

Reply:
`-Xcomp` is always very slow (after all it compiles everything), unless you restrict compilation.
So here, I would remove both `-Xcomp` and the compilation restriction. That will make the run fast again. But then it is important to have enough repetitions of your test method, see comment lower down.

Reply:
Done, both flags removed now (and number of iterations increased).